packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,27 +8,43 @@ import { InitError, SupportError } from './errors/index.mjs'
  *
  * @internal
  * @virtual
+ * @template [T=HTMLElement]
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @protected
+   * @type {T}
+   */
+  $root
+
   /**
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
-   * @param {Element | null} [$root] - HTML element to use for component
+   * @param {T | null} [$root] - HTML element to use for component
```

Comment:
**issue** Could this be kept as `Element` as well, for a similar reason as the comment on the `Accordion` class.